cmd/dump/src/lib.rs

```diff
@@ -758,8 +1020,35 @@ fn dump_via_agent(
     subargs: &DumpArgs,
 ) -> Result<()> {
     let mut agent = AgentCore::new(hubris)?;
-    let regions = hubris.regions(core)?;
     let started = Some(Instant::now());
+    let mut area = subargs.area.map(DumpArea::ByIndex);
+
+    let mut context = HiffyContext::new(hubris, core, subargs.timeout)?;
+    let funcs = context.functions()?;
+
+    //
+    // Our task can come from a couple of different spots:  we can either
+    // be explicitly told our task (in which case we are simulated or
+    // emulated) or we are pulling it out of an area.  If the latter, we
+    // will discover the task when we actually read our dump headers, so
+    // leave it as None for now...
+    //
+    let mut task = match &subargs.task {
+        Some(task) => {
+            let ndx = match hubris.lookup_task(task) {
+                Some(HubrisTask::Task(ndx)) => ndx,
+                _ => {
+                    bail!("invalid task \"{task}\"");
+                }
+            };
+
+            Some(DumpTask::new(
+                *ndx as u16,
+                core.read_word_64(hubris.lookup_variable("TICKS")?.addr)?,
```

Comment:
You could use the new `hubris.ticks` here!

Reply:
Ha, good point!